server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -511,9 +511,17 @@ public void reportShuffleResult(
               + partitionToBlockIds.size()
               + " blocks as shuffle result for the task of "
               + requestInfo);
-      shuffleServer
-          .getShuffleTaskManager()
-          .addFinishedBlockIds(appId, shuffleId, partitionToBlockIds, bitmapNum);
+      int duplicatedBlockIdNum =
+          shuffleServer
+              .getShuffleTaskManager()
+              .addFinishedBlockIds(appId, shuffleId, partitionToBlockIds, bitmapNum);
+      if (duplicatedBlockIdNum > 0) {
```

Comment:
`addFinishedBlockIds` return added blocks size. We can get total blocks size by partitionToBlockIds. We can use them to calculate the duplicate blocks size.

Reply:
Done